### ISSUE TITLE
Implement an approach to pass Cognito tokens from JupyterHub to JupyterLab

### ISSUE THREAD

**Issue by ramesh-maddegoda** [unity-sds/unity-cs#129]

**Comment by ramesh-maddegoda** [comment]
Work in progress code is available at https://github.com/unity-sds/unity-cs-security/tree/jupyterhub-custom-authenticator/code_samples/jupyter/custom-authenticator 

- Implemented a custom OAuthenticator to pass Cognito tokens from JupyterHub to JupyterLab
- Updated the custom authenticator to store the access token expiry time, and also added a markdown file representing a Jupyter Notebook to show a way to refresh access tokens

**Comment by galenatjpl** [comment]
@ramesh-maddegoda is #130 still blocking this?

**Comment by ramesh-maddegoda** [comment]
@galenatjpl , no, I was able to completely implement and test this on local environment.

**Comment by galenatjpl** [comment]
@ramesh-maddegoda ok great!  So is this complete now, or are we waiting to close once we do more full testing and integration into the cloud-deployed ADS?

**Comment by ramesh-maddegoda** [comment]
@galenatjpl , we can consider this as complete now. Because this was very specific to setting up a development environment on MCP, which is not required anymore. 

If this issue also exists with cloud-deployed ADS, then their current JupyterHub will also not work (even before integrating our code). In that case, it should be handled with a different ticket.
